source/dataman/DataMan.cpp

```diff
@@ -58,14 +59,14 @@ void DataMan::add_stream(json a_jmsg)
 
     if (m_tolerance.size() < m_num_channels)
     {
-        for (int i = 0; i < m_num_channels; i++)
+        for (int i = 0; i < m_num_channels; ++i)
```

Comment:
You can replace this loop entirely with `m_tolerance.insert(m_tolerance.end(), m_num_channels, 0);` to insert `m_num_channels` copies of `0`